internal/alloydbutil/engine.go

```diff
@@ -138,9 +138,47 @@ func getServiceAccountEmail(ctx context.Context) (string, error) {
 	return userInfo.Email, nil
 }
 
+// NewVectorstoreTableOptions initializes the options struct with the default values for
+// the InitVectorstoreTable function.
+func NewVectorstoreTableOptions(opts *VectorstoreTableOptions) (*VectorstoreTableOptions, error) {
+	vectorstoreTableOptions := new(VectorstoreTableOptions)
+	if opts.TableName == "" {
+		return vectorstoreTableOptions, fmt.Errorf("missing table name in options")
+	}
+	if opts.VectorSize == 0 {
+		return vectorstoreTableOptions, fmt.Errorf("missing vector size in options")
+	}
+	vectorstoreTableOptions.TableName = opts.TableName
+	vectorstoreTableOptions.VectorSize = opts.VectorSize
+	if opts.SchemaName != "" {
+		vectorstoreTableOptions.SchemaName = opts.SchemaName
+	} else {
+		vectorstoreTableOptions.SchemaName = "public"
+	}
+
+	if opts.ContentColumnName != "" {
+		vectorstoreTableOptions.ContentColumnName = opts.ContentColumnName
+	} else {
+		vectorstoreTableOptions.ContentColumnName = "content"
+	}
+
+	if opts.EmbeddingColumn != "" {
+		vectorstoreTableOptions.EmbeddingColumn = opts.EmbeddingColumn
+	} else {
+		vectorstoreTableOptions.EmbeddingColumn = "embedding"
+	}
+
+	if opts.MetadataJsonColumn != "" {
+		vectorstoreTableOptions.MetadataJsonColumn = opts.MetadataJsonColumn
+	} else {
+		vectorstoreTableOptions.MetadataJsonColumn = "langchain_metadata"
+	}
+
+	return vectorstoreTableOptions, nil
+}
+
 // initVectorstoreTable creates a table for saving of vectors to be used with PostgresVectorStore.
-func (p *PostgresEngine) InitVectorstoreTable(ctx context.Context, tableName string, vectorSize int, schemaName string, contentColumn string,
-	embeddingColumn string, metadataColumns []Column, metadataJsonColumn string, idColumn Column, overwriteExisting bool, storeMetadata bool) error {
+func (p *PostgresEngine) InitVectorstoreTable(ctx context.Context, vsTableOpts VectorstoreTableOptions, metadataColumns []Column, overwriteExisting bool, storeMetadata bool) error {
```

Comment:
This function signature seems wrong. Shouldn't metadataColumns, overwriteExisting, and storeMetadata also be optional? And thus part of the options struct?